### ISSUE TITLE
for NER downstream task, the data format of train.csv/test.csv should be different , right?

### ISSUE THREAD

**Issue by marcusau** [PUSSYMIPT/bert-distillation#6]
Hi ,

   Highly appreciated your amazing work.

   The dataset in your data folder are for text classification purpose.

   What is the data format and data preprocessing for NER task?

Thanks a lot

Marcus

**Comment by PUSSYMIPT** [comment]
Hi, @marcusau! 
In this repository, I use the dataset for masked language modelling task. The example in the `data` folder is just for fast testing in CI. 

**So, for now, this project only support distillation for LM task.** 

The main idea is that if you have enough time, large corpus with your language and big pre-trained BERT model for your language (for example [here](https://huggingface.co/models)) you can distillate it to smaller model and then fine-tune it for any task you want (Classification, NER, question answering etc). For example, for the Russian language, there is a big BERT model from DeepPavlov, but there is no Russian DistilBERT, so you can use this repository to create one.
If someone already distillate BERT for your language you can just load it and then fine-tune it for your task.

If you still want to do distillation for your task probably you could change code in `src/models` and use `DistilBertForSequenceClassification` and `BertForSequenceClassification` instead of `DistilBertForMaskedLM` and `BertForMaskedLM`, but I can't guarantee that it will work. Also, I'm not sure about input format for NER task in Hugging Face transformers library. All I can do is just refer to hf [docs](https://huggingface.co/transformers/index.html). Hope this information will help you!

Nikita

**Comment by marcusau** [comment]
so if i have huge corpus, then i can follow your codes and distillate it to smaller model for my downstream tasks ,right?

If my dataset has 120000 news articles with about ~3million sentences, this corpus is huge enough yet?

Thanks a lot.

**Comment by elephantmipt** [comment]
Well, can you tell about your task more specific? What is the language of your dataset? What model are you going to use as a teacher? 

If you need your model only for NER task I would recommend you to try distillate it directly for NER task, but there is no way to do so with this repository. The next option is to distillate your model with LM task and then fine-tune it for NER, this will take more time and resources, but you could reuse model for other tasks like classification, QA and so on. Again, you can't use this code to distillate model for NER task directly.

**Comment by marcusau** [comment]
language of your dataset: Financial news about Hong Kong + China in "traditional Chinese"

Tasks:
phrase 1 tasks: POS + NER
phrase 2 tasks: Knowledge graph building + relationship parser

This is why i expect my BERT distillation could be multi-task learning at the end of this project.

Ofz at the beginning, the distillation could be focused only on NER duties.

**Comment by elephantmipt** [comment]
Okay, you can pass something like this: https://gist.github.com/elephantmipt/4287f5792a4c1e716d2f62db623646cf . Don't forget to specify path to your dataset and text_field in config above. You can run it with 

`catalyst-dl run -C PATH_TO_CONFIG --verbose`

Please feedback if something will fail.

**Comment by marcusau** [comment]
Thanks a lot.. i will try with this
